### PR TITLE
Improve docs about the compilation database

### PR DESCRIPTION
**Pull request by Discookie**
Added documentation about `make` and `cmake` to the extension's README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +39,17 @@ This is a C/C++ code analysis plugin for VSCode that shows bug reports detected
 
 ![Commands and tasks](media/codechecker-tasks.gif)
 
+## Setting up your build environment
+
+CodeChecker works by reading a generated compilation database, often called `compile_commands.json`, and executing specific analyses based on that.
```
Comment (dkrupp) [inline README.md:44]: You may refer to the description of the compilation database https://clang.llvm.org/docs/JSONCompilationDatabase.html
so users understand better what this is about.

**Review by dkrupp — CHANGES_REQUESTED** [review]
Just a minor comment: please referer to the compilation datbase format. otherwise lgtm.

File: README.md (hunk 1 above)
Comment (Discookie) [reply README.md]: Good to know there's a link for the file format! Added.

**Review by MiklosMagyari — APPROVED** [review]
lgtm

**Review by dkrupp — APPROVED** [review]

**Pull request merged**